### PR TITLE
Fix two small issues with protocol upgrades

### PR DESCRIPTION
**Pull request by mystenmark**
Change the upgrade threshold to be purely stake based, make the additional stake configurable as a fraction of f, so that upgrades proceed if 2f+1 + B*f vote, where B is configured in ProtocolConfig.

### CODE REVIEW

File: crates/sui-core/src/authority.rs [hunk 1]
```diff
@@ -2549,6 +2545,65 @@ impl AuthorityState {
...
 
+    fn choose_next_protocol_version(epoch_store: &Arc<AuthorityPerEpochStore>) -> ProtocolVersion {
+        let protocol_config = epoch_store.protocol_config();
+        let committee = epoch_store.committee();
+
+        // Determine which protocol version to propose for the following epoch.
+        let current_protocol_version = committee.protocol_version;
+        let next_protocol_version = current_protocol_version + 1;
+
+        let mut stake_aggregator: StakeAggregator<bool, true> =
```
Comment (lxfind) [inline crates/sui-core/src/authority.rs:2556]: You can use () instead of bool

File: crates/sui-protocol-config/src/lib.rs [hunk 2]
```diff
@@ -490,6 +500,8 @@ impl ProtocolConfig {
                 stake_subsidy_rate: Some(1),
                 storage_gas_price: Some(1),
                 max_transactions_per_checkpoint: Some(1000),
+                // require 2f+1 + 0.75 * f stake for automatic protocol upgrades.
+                buffer_stake_for_protocol_upgrade_bps: Some(7500),
```
Comment (lxfind) [inline crates/sui-protocol-config/src/lib.rs:504]: Please move 7500 to protocol constants file and add a TODO to tune it

Comment (mystenmark) [reply crates/sui-protocol-config/src/lib.rs:504]: This is the protocol constants file.

Comment (lxfind) [reply crates/sui-protocol-config/src/lib.rs:505]: Oh didn't know it got renamed

**Review by lxfind — APPROVED** [review]

File: crates/sui-protocol-config/src/lib.rs (hunk 2 above)
Comment (mystenmark) [reply crates/sui-protocol-config/src/lib.rs:505]: yeah, because they aren't really constants anymore - the can change from one version to the next.

**Pull request merged**